api/submission/fetch_course_report.go

```diff
@@ -0,0 +1,30 @@
+package submission
+
+import (
+	"github.com/edulinq/autograder/api/core"
+	"github.com/edulinq/autograder/report"
+)
+
+type FetchCourseReportRequest struct {
+    core.APIRequestCourseUserContext
+    core.MinRoleAdmin
+}
+
+type FetchCourseReportResponse struct {
+	CourseReport *report.CourseScoringReport `json:"course-report"`
+}
+
+func HandleFetchCourseReport(request *FetchCourseReportRequest) (*FetchCourseReportResponse, *core.APIError) {
+	response := FetchCourseReportResponse{};
```

Comment:
Construct this closer to where it is used.
You can also (optionally) just construct it on the return (since this response is simple).